src/webapp/components/multiple-selector/useMultipleSelector.tsx

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { useBooleanState } from "$/webapp/utils/use-boolean";
+import { MultipleSelectorProps } from "$/webapp/components/multiple-selector/MultipleSelector";
+import i18n from "$/utils/i18n";
+import _ from "$/domain/entities/generic/Collection";
+
+export function useMultipleSelector(props: MultipleSelectorProps) {
+    const {
+        items,
+        values,
+        onChange,
+        allOption,
+        customMenu,
+        type = "item",
+        pluralType = "items",
+    } = props;
+
+    const [menuIsOpen, { enable: openMenu, disable: closeMenu }] = useBooleanState(false);
+    const [tooltipIsOpen, { enable: openTooltip, disable: closeTooltip }] = useBooleanState(false);
+    const [filterText, setFilterText] = React.useState("");
+
+    const notifyChange = React.useCallback(
+        (event: React.ChangeEvent<{ value: unknown }>) => {
+            onChange(
+                (event.target.value as string[]).filter(
+                    s => s !== "multiple-selector-void" && s !== undefined && s !== null
```

Comment:
There is a mismatch between the type you force (`string[]`) and the things that you check